Dockerfile

```diff
@@ -1,22 +1,24 @@
-FROM rust:1.54.0-alpine as builder
+FROM rust:1.54.0-alpine as targetarch
 
-# Check cargo version
-RUN cargo --version
+ARG BUILDPLATFORM
+ARG TARGETPLATFORM
+ARG TARGETARCH
 
-# Download binaryen and verify checksum
-ADD https://github.com/WebAssembly/binaryen/releases/download/version_96/binaryen-version_96-x86_64-linux.tar.gz /tmp/binaryen.tar.gz
-RUN sha256sum /tmp/binaryen.tar.gz | grep 9f8397a12931df577b244a27c293d7c976bc7e980a12457839f46f8202935aac
+RUN echo "Running on $BUILDPLATFORM, building for $TARGETPLATFORM"
 
-# Extract and install wasm-opt
-RUN tar -xf /tmp/binaryen.tar.gz
-RUN mv binaryen-version_*/wasm-opt /usr/local/bin
+FROM targetarch as builder-amd64
+ARG ARCH="x86_64"
 
-# Check wasm-opt version
-RUN wasm-opt --version
+FROM targetarch as builder-arm64
+ARG ARCH="aarch64"
+
+FROM builder-${TARGETARCH} as builder
```

Comment:
This is clever 👍

Reply:
Yeah, I had to introduce that because the machine and arch names don't match.

I think it's better / clearer than an if / test construct. It can also server as a place where to put extra arch-specific stuff in the future.